TombEngine/Scripting/Internal/TEN/Objects/Moveable/MoveableObject.cpp

```diff
@@ -1016,3 +1047,96 @@ bool Moveable::MeshExists(int index) const
 
 	return true;
 }
+
+//Attach camera and camera target to a mesh of an object.
+void  Moveable::AttachCamera(short camMeshId, short targetMeshId)
+{
+	ObjCamera(m_item, camMeshId, m_item, targetMeshId, true);
+}
+
+//Lara will undraw her weapons if they are drawn, throw away flare if she helds one in her hand.
+void   Moveable::UndrawWeapons()
+{
+	m_item->IsLara();
```

Comment:
This code does nothing, because it just calls bool check which returns result into nothing. You should add a condition for it: `if (!m_item->IsLara()) return`.

But anyway, as we discussed in Discord, all `Lara`-specific lua functions should be contained inside `Lara` wrapper class (similar to `Moveable::...` functions, but they should then start with `Lara::...`). @goldlife2904 is currently working on it, so I propose to wait for his changes and then convert all Lara functions to it.

Reply:
yes, a new Lara class would be great, then I can move all Lara-specific codes from moveable to lara wrapper class.